### DISCUSSION TITLE
velocity curve in fluidsynth

### DISCUSSION THREAD

**Discussion by ghost** [FluidSynth/fluidsynth#988] [General]

**Comment by mawe42** [comment]
Standard transfer functions from raw controller value to volume are logarithmic in nature in virtually all MIDI synths. The main reason is that human perception of volume is not linear but closer to a logarithmic as well. I would be surprised if a linear transfer function would yield the desired effect on your MIDI keyboard. Anyway... a Google search with "midi volume logarithmic" or similar keywords will give you more information on the topic.

Having said that, SoundFonts allow you to change the transfer functions for controllers. So if you really want a linear response, you could edit a SoundFont with something like Polyphone to achieve the desired result.

**Reply by mawe42** [reply]
You could read the SoundFont specs: http://www.synthfont.com/SFSPEC21.PDF
Particularly sections 8.4.1 MIDI Note-On Velocity to Initial Attenuation, the definition of "centibels" and 9.5.2 Pictorial Examples of Source Types.

This paper might also be of interest: https://www.cs.cmu.edu/~rbd/papers/velocity-icmc2006.pdf
